### PR TITLE
OBSDOCS-1034/TRACING-4069: Write docs for the filelog receiver component

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is tied to a product release date, which is currently set to **June 5** (Wednesday).

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

My edit of https://github.com/openshift/openshift-docs/pull/73770

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/OBSDOCS-1034
https://issues.redhat.com/browse/TRACING-4069

Link to docs preview:

https://75550--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector.html#filelog-receiver_otel-configuration-of-otel-collector

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by IshwarKanse — APPROVED** [review]

**Review by rubenvp8510 — APPROVED** [review]

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -199,6 +199,32 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
 Wildcards with `+*+` are accepted under the `cors_allowed_origins`.
 To match any origin, enter only `+*+`.
 
+[id="filelog-receiver_{context}"]
+=== Filelog Receiver
+
+:FeatureName: The Filelog Receiver
+include::snippets/technology-preview.adoc[]
+
+The Filelog Receiver tails and parses logs from files.
```
Comment (bscott-rh) [inline modules/otel-collector-components.adoc:208]:
```suggestion
The Filelog Receiver tails and parses logs from files.

:FeatureName: The Filelog Receiver
include::snippets/technology-preview.adoc[]
```
I would recommend putting the abstract/introductory sentence before the tech preview snippet. it's useful to get some information about what the component is before reading about its support status.

**Review by bscott-rh — COMMENTED** [review]
I left one suggestion for your consideration. otherwise, LGTM

File: modules/otel-collector-components.adoc (hunk 1 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:208]: Thank you. I'll swap these lines as part of a global revision PR for this file. There are more lines like these throughout the file, and those are outside of this PR, so I'll fix it globally for the whole file in a single PR.

**Comment by max-cx** [comment]
Closing this PR because I cherry-picked its commit into https://github.com/openshift/openshift-docs/pull/76340.

**Comment by kcarmichael08** [comment]
looks good to merge when product is shipped

**Pull request merged**